node-stringprep.cc

```diff
@@ -270,28 +269,29 @@ NAN_METHOD(ToASCII)
       {
         // other error, just bail out
         delete[] dest;
-        NanThrowError(u_errorName(error));
-        NanReturnUndefined();
+        Nan::ThrowError(u_errorName(error));
+        return;
       }
 
-    Local<String> result = NanNew<String>(dest, destLen);
+    Local<String> result = Nan::New<String>(dest, destLen).ToLocalChecked();
     delete[] dest;
-    NanReturnValue(result);
+    info.GetReturnValue().Set(result);
   }
   else {
-    NanThrowTypeError("Bad argument.");
-    NanReturnUndefined();
+    Nan::ThrowTypeError("Bad argument.");
+    return;
   }
 }
 
 /*** Initialization ***/
 
-extern "C" void init(Handle<Object> target)
-{
-  NanScope();
-  StringPrep::Initialize(target);
-  NODE_SET_METHOD(target, "toUnicode", ToUnicode);
-  NODE_SET_METHOD(target, "toASCII", ToASCII);
+extern "C" {
+  static void init (Handle<Object> target)
```

Comment:
`Handle` will end up being a problem here (it's being deprecated so you'll end up with errors in a version of Node or two), opt for declaring this as `NAN_MODULE_INIT(init)` anyway and you'll be safe